cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -381,8 +381,7 @@ public void resumeLoadMonitorActivity() {
    */
   public void executeProposals(Collection<ExecutionProposal> proposals,
                                Collection<Integer> unthrottledBrokers) {
-    // Pause the load monitor before the proposal execution.
-    _loadMonitor.pauseMetricSampling();
+    // Add execution proposals and start execution.
```

Comment:
Looking at the code. If there are two threads calling executeProposals() with two different set of proposals, CC may execute the both sets of proposals. Maybe we should just let the executor expose a method of `executeProposals()` and synchronize on that.

Reply:
Made the Executor thread-safe.